src/ng/interpolate.js

```diff
@@ -301,27 +307,54 @@ function $InterpolateProvider() {
         }
       }
 
+      if (concat.length === 1 && expressionPositions.length === 1) {
+        singleExpression = true;
+      }
+
       // Concatenating expressions makes it hard to reason about whether some combination of
       // concatenated values are unsafe to use and could easily lead to XSS.  By requiring that a
-      // single expression be used for iframe[src], object[src], etc., we ensure that the value
-      // that's used is assigned or constructed by some JS code somewhere that is more testable or
-      // make it obvious that you bound the value to some user controlled value.  This helps reduce
-      // the load when auditing for XSS issues.
-      if (trustedContext && concat.length > 1) {
-          $interpolateMinErr.throwNoconcat(text);
-      }
```

Comment:
Why remove the checks at this point and defer them inside `compute()` instead?
Are we gaining anything?